src/magic.cpp

```diff
@@ -1066,15 +1067,33 @@ void spell::cast_spell_effect( Creature &source, const tripoint &target ) const
 
 void spell::cast_all_effects( Creature &source, const tripoint &target ) const
 {
-    // first call the effect of the main spell
-    cast_spell_effect( source, target );
-    for( const fake_spell &extra_spell : type->additional_spells ) {
-        spell sp = extra_spell.get_spell( extra_spell.level );
+    if( has_flag( spell_flag::WONDER ) ) {
+        const auto iter = type->additional_spells.begin();
+        for( int num_spells = abs( damage() ); num_spells > 0; num_spells-- ) {
+            const int rand_spell = rng( 0, type->additional_spells.size() - 1 );
```

Comment:
Is it safe to assume that `type->additional_spells` is always non-empty? Because this will cause UB when it's empty.

Reply:
yeah, it should always be non-empty due to the flag changing the behavior, though an error message would probably be ideal.

Reply:
Have you tested what happens when "extra_effects" is an empty array in the "AEA_BUGS" spell (**keeping** the "WONDER" flag)? Will it show an error?

Reply:
seg fault. i'll make a PR with some error handling.